### DISCUSSION TITLE
creation of .env file for use in local development by vercel cli

### DISCUSSION THREAD

**Discussion by candh** [vercel/vercel#5244] [Help] [answered]
`vercel env pull` creates `.env` file which is [not ignored](https://vercel.com/docs/build-step?query=ignore#ignored-files-and-folders) by Vercel by default.

Shouldn't the cli create `.env.local` file so that it doesn't get accidentally uploaded to the server and goes unnoticed?

**Comment by styfle** [comment, marked as answer]
Hi @candh 
It depends on your framework, not all support `.env.local`.

If you're using Next.js, we suggest using `vercel env pull .env.local` and ignoring it.

https://nextjs.org/docs/basic-features/environment-variables#environment-variables-on-vercel

In the future, we might [change the default to `.env.local`](https://github.com/vercel/next.js/issues/16763) and automatically ignore this file when you pull it.

Thanks!

**Comment by candh** [comment]
Thank you for clearing this up.
